Libplanet/Misc/ArrayEqualityComparer.cs

```diff
@@ -0,0 +1,99 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+
+namespace Libplanet.Misc
+{
+    /// <summary>
+    /// An <see cref="IEqualityComparer{T}"/> implementation to compare two arrays of the same
+    /// element type.  This compares the elements in the order of the array.
+    /// <para>The way to compare each element can be customized by specifying
+    /// the <see cref="ElementComparer"/>.</para>
+    /// </summary>
+    /// <typeparam name="T">The element type of the array.</typeparam>
+    public class ArrayEqualityComparer<T> : IEqualityComparer<T[]>
```

Comment:
Why did you build comparer instead of using [`Enumerable.SequenceEqual()`](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.sequenceequal?view=net-6.0)?

Reply:
Because `ConcurrentDictionary<K, V>` requires an `IEqualityComparer<T>` implementation.  It is used for accurate matching of `byte[]` keys.  (.NET arrays do not implement `IEquatable<T>`.)

Reply:
This looks a bit too generic and possibly dangerous. 😶 

Arrays and lists are base enough classes where it is expected that `IEquatable<T>` isn't implemented, i.e. `new int[] { 0 } != new int[] { 0 }`. Such high visibility, access level `public`, and scope, namespace suggesting global utility `class`, makes this prone to misuses.

Case in point, `Commit(INode node, IDictionary<byte[], byte[]> values)` has no idea whether a given `IDictionary` will be using the said comparer. Looking at the code locally, a reader might be easily confused. I don't know the best way mitigate such possible issue, but I can make a few suggestions.

- Lowering visibility and/or access level.
- Wrap `byte[]` in a new class implementing `IEquatable<T>`.
- Wrap `Dictionary<TKey, TValue>` in a new class with default comparer as `ArrayEqualityComparer<T>`.

Or you can just put this on a backlog since it isn't that critical. 🙃

Reply:
I believe this (and any other `IEqualityComparer<T[]>`) is not that prone to misuse, because the interface in itself is not widely used and only used in a narrow scope — mostly for dictionaries. 🤔 

In general, using arrays as dictionary keys are not a good idea.  It was already designed in that way before I kicked this off, and I left it as is due to in the interest of time… 😭 Anyway, I am going to define a distinct type satisfying `IEquatable<T>` and replace arrayed keys with it in the future.